mlrun/runtimes/nuclio/function.py

```diff
@@ -274,6 +288,37 @@ def pre_deploy_validation(self):
         if self.metadata.tag:
             mlrun.utils.validate_tag_name(self.metadata.tag, "function.metadata.tag")
 
+    def mask_sensitive_data_in_config(self):
+        if not self.spec.config:
+            return {}
+
+        raw_config = copy.deepcopy(self.spec.config)
+
+        for key, value in self.spec.config.items():
+            if key.startswith("spec.triggers"):
```

Comment:
Since you enlightened us with the list in https://github.com/nuclio/nuclio/blob/01b66be40410cb9827b3de0eb7c8628ade18911b/pkg/platformconfig/types.go#L305

what about the other fields in the list? should we consider them as well?

Reply:
@rokatyy the list can also change, as it accepts custom sensitive fields as an input to the platformconfig.

Reply:
@TomerShor yes, list can change, however we don't have a sync mechanism for now between mlrun and nuclio, so we had either this option or an option of duplicating secret creation mechanism in mlrun.

So it was decided to go with this option, assuming that fields won't change and masking is always enabled.

Reply:
@alonmr from all other params, I'd say it would be also nice to support codeentryattributes masking (because I found places where we fill them in with sensitive data), volumes in mlrun are already masked from what I understand. 

And annotations are up to us, but can be added as well

Reply:
@rokatyy this sounds like a change request we can add